// @iflow-mcp/borgius-jobspy-mcp-server/src/schemas/searchParamsSchema.js
export const searchParams = {
  siteNames: {
    type: 'string',
    description: 'Comma-separated list of job sites to search. Options: indeed,linkedin,zip_recruiter,glassdoor,google,bayt,naukri',
    default: 'indeed',
  },
  searchTerm: {
    type: 'string',
    description: 'Search term for jobs',
    default: 'software engineer',
  },
  location: {
    type: 'string',
    description: 'Location for job search',
    default: 'remote',
  },
  distance: {
    type: 'number',
    description: 'Distance in miles',
    default: 50,
  },
  jobType: {
    type: 'string',
    description: 'Type of job: fulltime, parttime, internship, contract',
    enum: ['fulltime', 'parttime', 'internship', 'contract'],
  },
  googleSearchTerm: {
    type: 'string',
    description: 'Google specific search term',
  },
  resultsWanted: {
    type: 'number',
    description: 'Number of results wanted',
    default: 20,
  },
  easyApply: {
    type: 'boolean',
    description: 'Filter for jobs that are hosted on the job board site',
    default: false,
  },
  descriptionFormat: {
    type: 'string',
    description: 'Format type of the job descriptions',
    enum: ['markdown', 'html'],
    default: 'markdown',
  },
  offset: {
    type: 'number',
    description: 'Starts the search from an offset',
    default: 0,
  },
  hoursOld: {
    type: 'number',
    description: 'How many hours old the jobs can be',
    default: 72,
  },
  verbose: {
    type: 'number',
    description: 'Controls verbosity (0=errors only, 1=errors+warnings, 2=all logs)',
    default: 2,
  },
  countryIndeed: {
    type: 'string',
    description: 'Country for Indeed search',
    default: 'USA',
  },
  isRemote: {
    type: 'boolean',
    description: 'Whether to search for remote jobs only',
    default: false,
  },
  linkedinFetchDescription: {
    type: 'boolean',
    description: 'Whether to fetch LinkedIn job descriptions (slower)',
    default: true,
  },
  linkedinCompanyIds: {
    type: 'string',
    description: 'Comma-separated list of LinkedIn company IDs',
  },
  enforceAnnualSalary: {
    type: 'boolean',
    description: 'Converts wages to annual salary',
    default: false,
  },
  proxies: {
    type: 'string',
    description: 'Comma-separated list of proxies',
  },
  caCert: {
    type: 'string',
    description: 'Path to CA Certificate file for proxies',
  },
  format: {
    type: 'string',
    description: 'Output format',
    enum: ['json', 'csv'],
    default: 'json',
  },
  timeout: {
    type: 'number',
    description: 'Timeout in milliseconds for the job search process',
    default: 120000,
  },
};
